chips/sam4l/src/usart.rs

```diff
@@ -591,26 +803,26 @@ impl hil::uart::UART for USART {
 
         // set USART mode register
         let mut mode = 0x00000000;
-        mode |= 0x1 << 19; // OVER: oversample at 8 times baud rate
-        mode |= 0x3 << 6; // CHRL: 8-bit characters
-        mode |= 0x0 << 4; // USCLKS: select CLK_USART
 
-        match params.stop_bits {
-            hil::uart::StopBits::One => mode |= 0x0 << 12, // NBSTOP: 1 stop bit
-            hil::uart::StopBits::Two => mode |= 0x2 << 12, // NBSTOP: 2 stop bits
+        mode |= Mode::OVER.val(1).to_u32(); // OVER: oversample at 8x
+        mode |= Mode::CHRL::BITS8.to_u32(); // CHRL: 8-bit characters
+        mode |= Mode::USCLKS::CLK_USART.to_u32(); // USCLKS: select CLK_USART
+
+        mode |= match params.stop_bits {
+            hil::uart::StopBits::One => Mode::NBSTOP::BITS_1_1.to_u32(),
+            hil::uart::StopBits::Two => Mode::NBSTOP::BITS_2_2.to_u32(),
```

Comment:
Instead of dealing with integer values in this section, you can sum bits.

```
let mode = Mode::OVER.SET;
mode += Mode::CHRL::BITS8;
mode += Mode::USCLKS::CLK_USART;

mode += match params.stop_bits {
    hil::uart::StopBits::One => Mode::NBSTOP::BITS_1_1,
    hil::uart::StopBits::Two => Mode::NBSTOP::BITS_2_2,
}

...
```